src/react/backend/location/__tests__/Locations.test.tsx

```diff
@@ -105,121 +126,90 @@ describe('Locations', () => {
-    expect(fourthRowColumns[2]).toEqual('');
-    // edit button
-    expect(
-      fourthRow.find(T.Cell).find(T.ActionButton).first().prop('disabled'),
-    ).toBeFalsy();
-    // delete button
-    expect(
-      fourthRow.find(T.Cell).find(T.ActionButton).at(1).prop('disabled'),
-    ).toBeFalsy();
-    const fifthRow = rows.at(4);
-    const fifthRowColumns = fifthRow
-      .find(T.Cell)
-      .map((column) => column.text());
-    expect(fifthRowColumns.length).toEqual(nbrOfColumnsExpectedWithoutXDM);
-    expect(fifthRowColumns[0]).toEqual('location-nfs');
-    expect(fifthRowColumns[1]).toEqual('NFS Mount');
-    expect(fifthRowColumns[2]).toEqual('');
-    // edit button
-    expect(
-      fifthRow.find(T.Cell).find(T.ActionButton).first().prop('disabled'),
-    ).toBeFalsy();
-    // delete button
-    expect(
-      fifthRow.find(T.Cell).find(T.ActionButton).at(1).prop('disabled'),
-    ).toBeFalsy();
-    const sixthRow = rows.at(5);
-    const sixthRowColumns = sixthRow
-      .find(T.Cell)
-      .map((column) => column.text());
-    expect(sixthRowColumns.length).toEqual(nbrOfColumnsExpectedWithoutXDM);
-    expect(sixthRowColumns[0]).toEqual('location-ring');
-    expect(sixthRowColumns[1]).toEqual('Scality RING with S3 Connector');
-    expect(sixthRowColumns[2]).toEqual('bucketName3');
-    // edit button
-    expect(
-      sixthRow.find(T.Cell).find(T.ActionButton).first().prop('disabled'),
-    ).toBeFalsy();
-    // delete button
-    expect(
-      sixthRow.find(T.Cell).find(T.ActionButton).at(1).prop('disabled'),
-    ).toBeFalsy();
-    const seventhRow = rows.at(6);
-    const seventhRowColumns = seventhRow
-      .find(T.Cell)
-      .map((column) => column.text());
-    expect(seventhRowColumns.length).toEqual(nbrOfColumnsExpectedWithoutXDM);
-    expect(seventhRowColumns[0]).toEqual('location-sproxyd');
-    expect(seventhRowColumns[1]).toEqual('Scality RING with Sproxyd Connector');
-    expect(seventhRowColumns[2]).toEqual('');
-    // edit button
-    expect(
-      seventhRow.find(T.Cell).find(T.ActionButton).first().prop('disabled'),
-    ).toBeFalsy();
-    // delete button
-    expect(
-      seventhRow.find(T.Cell).find(T.ActionButton).at(1).prop('disabled'),
-    ).toBeFalsy();
+
+    expect(screen.getAllByRole('columnheader').length).toEqual(nbrOfColumnsExpectedWithoutXDM);
+    //console.log(prettyDOM(buttonOfFirstRow[1]))
+    /*********************************************************/
+    const firstRow = screen.getAllByRole('row')[1];
+    const gridCellOfFirstRow = getAllByRole(firstRow, 'gridcell');
+    const buttonOfFirstRow = getAllByRole(firstRow, 'button');
+
+    expectFirstRow(gridCellOfFirstRow, 0, 'location-aws-s3');
+    expectFirstRow(gridCellOfFirstRow, 1, 'Amazon S3');
+    expectFirstRow(gridCellOfFirstRow, 2, 'bucketName1');
+
+    expectButtonNotDisable(buttonOfFirstRow, 0, 'Edit Location');
+    expectButtonNotDisable(buttonOfFirstRow, 1, 'Delete Location');
```

Comment:
I suggest not selecting button by their location in the page but using accessible labels or displayed text, see https://github.com/scality/zenko-ui/pull/271